### PR TITLE
[CI]Implemented skip-if-unchanged

### PR DESCRIPTION
**Pull request by sushmita56**
## Description
This is an update on .drone.star file 
- skip pipeline based on changed code 

## Related Issue:
Closes https://github.com/owncloud/QA/issues/732
Closes https://github.com/owncloud/QA/issues/730

### When to run test pipelines
1. Changes only in:
```
"^.github/.*"
"^.vscode/.*",
"^changelog/.*",
"README.md",
".gitignore",
"CHANGELOG.md",
"CONTRIBUTING.md",
"COPYING",
"COPYING.documentation",
```
DO NOT RUN: `ctest` and `gui-test`

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -158,7 +160,7 @@ def build_and_test_client(ctx, c_compiler, cxx_compiler, build_type, generator,
             "os": "linux",
             "arch": "amd64",
         },
-        "steps": [
+        "steps": skipIfUnchanged(ctx, "unit_tests") + [
```
Comment (saw-jan) [inline .drone.star:163]:
```suggestion
        "steps": skipIfUnchanged(ctx, "unit-tests") + [
```

File: .drone.star [hunk 2]
```diff
@@ -205,7 +207,7 @@ def gui_tests(ctx, trigger = {}, depends_on = [], filterTags = [], version = "da
             "os": "linux",
             "arch": "amd64",
         },
-        "steps": [
+        "steps": skipIfUnchanged(ctx, "gui_tests") + [
```
Comment (saw-jan) [inline .drone.star:210]:
```suggestion
        "steps": skipIfUnchanged(ctx, "gui-tests") + [
```

File: .drone.star [hunk 3]
```diff
@@ -669,3 +671,43 @@ def githubComment(alternateSuiteName):
...
+    base_skip_steps = [
+        "^.github/.*",
+        "^changelog/.*",
+        "README.md",
+    ]
+
+    if type == "gui-tests":
+        gui_files = [
+            "^src/.*",
+        ]
```
Comment (Talank) [inline .drone.star:698]: I think gui test files are all inside `test` dir. Can you run gui tests when there is any change in either `test` or `src` dir. And then run unit tests when there is change in `src` dir?

**Comment by saw-jan** [comment]
Tested in https://drone.owncloud.com/owncloud/client/11479 and https://drone.owncloud.com/owncloud/client/11480

File: .drone.star [hunk 4]
```diff
@@ -688,3 +689,43 @@ def cancelPreviousBuilds():
...
+
+    skip = []
+    if type == "unit-tests":
+        skip = base + [
+            "^test/gui/.*",
+        ]
+
+    if type == "gui-tests":
+        skip = base + [
+            "^test/([^g]|g[^u]|gu[^i]).*",
```
Comment (saw-jan) [inline .drone.star:717]:
```suggestion
            "^test/([^g]|g[^u]|gu[^i])/.*",
```

**Review by Talank — APPROVED** [review]

File: .drone.star (hunk 4 above)
Comment (saw-jan) [reply .drone.star:717]: doesn't work though
previous one is correct

**Pull request merged**